### DISCUSSION TITLE
Problem with input mixer and memory stream

### DISCUSSION THREAD

**Discussion by joaquinmir** [pschatzmann/arduino-audio-tools#1325] [Q&A] [answered]
Hi! I have a problem with making a program that uses buttons to play samples. 

```
#include <AudioConfig.h>
#include <AudioLogger.h>
#include <AudioTools.h>
#include "AudioTools.h"
#include "hihat.h"
#include "kick.h"

AudioInfo info(44100, 1, 16);
I2SStream i2s;  // Output to I2S
MemoryStream music(hihat_raw, hihat_raw_len,false);
MemoryStream music2(kick_raw, kick_raw_len,false);
InputMixer<int16_t> mixer;
StreamCopy copier(i2s, mixer); // copies sound into i2s

void playButton(){
   music.begin();
}

void setup(){
  
    Serial.begin(115200);
    pinMode(19, INPUT_PULLDOWN);
    attachInterrupt(19, playButton, RISING);
    AudioLogger::instance().begin(Serial, AudioLogger::Info);
    mixer.add(music);
    mixer.add(music2);
    auto config = i2s.defaultConfig(TX_MODE);
    config.copyFrom(info);
    config.channels=1;
    config.sample_rate= 44100; 
    i2s.begin(config);
   
}

void loop(){
  copier.copy();
}
```

The button should play the sample from the memorystream named "music". The problem is that i'm not getting any sound. If I put the 2 samples to be played in the playButton() ISR the program works, but that is not what i'm looking for. Why is that happening?

**Comment by pschatzmann** [comment, marked as answer]
The copier is copying the mixer to i2s. If you just want to play music, you need to reconfigure it e.g. with copier.begin(i2s, music) in the playButton method. The following might work as well: you can reconfigure the mixer to ignore music2 by calling mixer.setWeight(2, 0)
